### ISSUE TITLE
MapFactory vs single-value enums

### ISSUE THREAD

**Issue by ArcturusMengsk** [jqno/equalsverifier#166]
# What steps will reproduce the problem?
See code.

# What is the code that triggers this problem?

    public static class Abcd
    {
        public enum Xyz
        {
            XYZ;
        }
        
        private Map<Xyz, String> myMap = new HashMap<>();

        @Override
        public int hashCode()
        {
            final int prime = 31;
            int result = 1;
            result = prime * result + ((myMap == null) ? 0 : myMap.hashCode());
            return result;
        }

        @Override
        public boolean equals(Object obj)
        {
            if (this == obj) return true;
            if (obj == null) return false;
            if (getClass() != obj.getClass()) return false;
            Abcd other = (Abcd) obj;
            return Objects.equals(myMap, other.myMap);
        }
    }
    
    @Test
    public void testAbcd()
    {
        EqualsVerifier.forClass(Abcd.class)
            .usingGetClass()
            .verify();
    }

# What error message or stack trace does EqualsVerifier give?
java.lang.AssertionError: Significant fields: equals does not use myMap, or it is stateless.
For more information, go to: http://www.jqno.nl/equalsverifier/errormessages
	at nl.jqno.equalsverifier.EqualsVerifier.handleError(EqualsVerifier.java:355)
	at nl.jqno.equalsverifier.EqualsVerifier.verify(EqualsVerifier.java:341)

# What did you expect?
I expected the test to pass, since myMap is part of equals() -- equals() and hashCode() were auto-generated in Eclipse.

# Which version of EqualsVerifier are you using?
2.2

# Please provide any additional information below.
It seems the problem lies in MapFactory class, when it creates the red and black Maps.  Since it only uses black objects as values, it relies on red and black keys being different.  But there is only one value in the enum, and therefore the red and black keys become the same object as well, giving 2 identical Maps.

(I haven't attempted to hack the enum to give it an additional value, though that would surely solve my problem... ;)  Thank you for that interesting read.)

**Comment by Stephan202** [comment]
Seem a fix a-la #157 should do :)

**Comment by jqno** [comment]
Thanks for reporting this! I'll take a look and try to fix it. I agree with @Stephan202 that the approach in #157 is likely the way to go. I'll fix it for List too, while I'm at it :).

Just out of curiosity: what's your use case for having a map with only one key-value pair? Seems like a simple variable would do :).

(I'm glad you enjoyed my blog post about the enums!)

**Comment by ArcturusMengsk** [comment]
I just joined this project, and found out some people were using a hashCode() and equals() verification framework developed in-house, whereas others were using jqno.EqualsVerifier.  Upon comparing code, I thought EqualsVerifier covered more stuff, so I attempted to change all verification to use this instead.  That's how I found this in issue, with one of the classes that was being tested by the other framework.  As for why it looks like this, I'm half-guessing, but it looks like this enum is something that could be extended with more values in the future.
